### PR TITLE
Define timeout retry and resolve_retries in resolver section

### LINKED ISSUE #172: timeout retry and resolve_retries in resolver section
In my haproxy configuration, I use dns resolver on legacy servers and I need to set 2 properties `timeout retry` and `resolve_retries` to manage these dns servers.
It should be very usefull to have these properties in resolver section (like `hold valid`).

### PR DESCRIPTION
**Pull request by ogrand**
fix #172

### CODE REVIEW

File: jobs/haproxy/spec [hunk 1]
```diff
@@ -369,6 +369,12 @@ properties:
   ha_proxy.dns_hold:
     description: "DNS Hold time"
     default: 10s
+  ha_proxy.timeout_retry:
```
Comment (domdom82) [inline jobs/haproxy/spec:372]: That property name seems a bit too generic. It should be clear that this timeout belongs to DNS resolution.
Following the pattern from other timeouts I suggest `resolve_retry_timeout`

File: jobs/haproxy/templates/haproxy.config.erb [hunk 2]
```diff
@@ -199,6 +199,8 @@ listen health_check_http_url
 <% if_p("ha_proxy.resolvers") do |resolvers| -%>
 resolvers default
     hold valid <%= p("ha_proxy.dns_hold") %>
+    timeout retry <%= p("ha_proxy.timeout_retry") %>
```
Comment (domdom82) [inline jobs/haproxy/templates/haproxy.config.erb:202]: `resolve_retry_timeout` see my comment above

File: spec/haproxy_templates_spec.rb [hunk 3]
```diff
@@ -62,6 +62,8 @@
         'forwarded_client_cert' => 'sanitize_set',
         'tcp' => [],
         'dns_hold' => '10s',
+        'timeout_retry' => '1s',
```
Comment (domdom82) [inline spec/haproxy_templates_spec.rb:65]: `resolve_retry_timeout` see my comment above

**Review by domdom82 — CHANGES_REQUESTED** [review]

**Comment by ogrand** [comment]
@domdom82 I changed name propertie to resolve_retry_timeout as your proposal. Thanks

**Comment by domdom82** [comment]
@geofffranks `failed to fetch digest for image 'gfranks/haproxy-boshrelease-testflight:latest': 429 Too Many Requests` we're gettin hit by the new dockerhub policy on rate limits XD

**Review by domdom82 — APPROVED** [review]

**Pull request merged**